pilot/pkg/kube/inject/mesh.go

```diff
@@ -22,10 +22,20 @@ const (
 	parameterizedTemplate = `
 [[- $proxyImageKey                  := "sidecar.istio.io/proxyImage" -]]
 [[- $interceptionModeKey            := "sidecar.istio.io/interceptionMode" -]]
+[[- $statusPortKey                  := "status.sidecar.istio.io/port" -]]
```

Comment:
My traditional comment - this file must be deleted, I believe it is only used for some tests because  
@mandarjog couldn't convert them in 0.8 - and we keep having to update and work on this.

AFAIK no production code should be using this - either a file or the configmap fetched from cluster are used.

Reply:
We would need a separate PR to get rid of that file, it is used extensively in the inject tests. Not sure we can get rid of it though, the templates seem used in injection @ostromart

Reply:
@costinm this comment is not particularly helpful and unrelated to this PR.  Please refrain from venting personal frustration in reviews.

Reply:
We can and should get rid of this built-in file, it is just work that keep getting punted on.
It should be separate PR though, and until then `parameterizedTemplate` should keep parity.